File: Week 1/greedy.c
#include <stdio.h>
#include <math.h>

int main(void){
  float f;
  	do
  	{
  	   printf(" How much change is owed? ");
       scanf("%f", &f);
  	} while (f <= 0);

    int coins = 0;
    int change = round(f * 100);

  	while (change > 0)
  	{
      if (change >= 25)
	    {
	        change -= 25;
	    }
	    else if (change >= 10)
	    {
	        change -= 10;
	    }
	    else if (change >= 5)
	    {
	        change -= 5;
	    }
	    else
	    {
	        change -= 1;
	    }
	    coins++;
  	}


  	printf("%i\n", coins);
}

File: Week 8/greedy.py
change = float(input("How much change is owed? "))
while change <= 0:
    change = float(input("Please enter a number bigger than 0"))

coins = 0;
change = round(change, 2)
while change > 0:
    if change >= 0.25:
        change -= 0.25    
    elif change >= 0.1:
        change -= 0.10
    elif change >= 0.05:
        change -= 0.05
    else:
	    change -= 0.01
    coins = coins + 1
print(coins)